src/core/unit.h

```diff
@@ -1078,6 +1078,15 @@ Condition *unit_find_failed_condition(Unit *u);
         ({                                                              \
                 const Unit *_u = (unit);                                \
                 const int _l = (level);                                 \
+                struct iovec *_input_iovec = NULL;                      \
+                size_t _n_input_iovec = 0;                              \
+                const ExecContext *_c = _u ?                            \
+                        unit_get_exec_context(_u) : NULL;               \
+                if (_c && unit_log_level_test(_u, _l)) {                \
+                        _input_iovec = _c->log_extra_fields;            \
+                        _n_input_iovec = _c->n_log_extra_fields;        \
+                }                                                       \
+                LOG_CONTEXT_PUSH_IOV(_input_iovec, _n_input_iovec);     \
                 (log_get_max_level() < LOG_PRI(_l) || (_u && !unit_log_level_test(_u, _l))) ? -ERRNO_VALUE(error) : \
```

Comment:
this doesn't look right. You do the unit_log_level_test() twice, but not the log_get_max_level(). 

Please do both, only once, and do not invoke unit_get_exec_context() unless they pass.